Remove "bad characters" from our codebase

potiuk opened the pull request:
We had plenty of "bad characters" in our codebase that were not
invited and came here by accident. We want to get rid of those
"bad characters" once and for all.

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

ashb commented on .gitmodules:
```diff
@@ -1,12 +1,12 @@
 [submodule ".github/actions/get-workflow-origin"]
-	path = .github/actions/get-workflow-origin
-	url = https://github.com/potiuk/get-workflow-origin
+    path = .github/actions/get-workflow-origin
```
Tabs are not bad characters -- flake8/black deals with this, so leave tabs alone.

ashb replied:
Or ignore the following files like we did before:

```
        exclude: ^airflow/_vendor/|^clients/gen/go\.sh$|^\.gitmodules$
```

potiuk replied:
we alredy had "forbid tabs" before - I just missed an exclusion (which I have to add anyway).

potiuk replied:
(this one will be better because it will fix them automatically).

potiuk committed:
Remove "bad characters" from our codebase

We had plenty of "bad characters" in our codebase that were not
invited and came here by accident. We want to get rid of those
"bad characters" once and for all.

ashb commented on scripts/ci/pre_commit/pre_commit_replace_bad_characters.py:
```diff
@@ -0,0 +1,73 @@
+if __name__ != "__main__":
+    raise Exception(
+        "This file is intended to be executed as an executable program. You cannot use it as a module."
+        "To run this script, run the ./build_docs.py command"
```
nit: copy pasta error here.
```suggestion
        f"To run this script, run the {__file__} command"
```

potiuk committed:
Update scripts/ci/pre_commit/pre_commit_replace_bad_characters.py

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>
